service/src/main/java/org/cbioportal/service/impl/AlterationDriverAnnotationServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package org.cbioportal.service.impl;
+
+import org.cbioportal.model.AlterationDriverAnnotation;
+import org.cbioportal.model.CustomDriverAnnotationReport;
+import org.cbioportal.persistence.AlterationDriverAnnotationRepository;
+import org.cbioportal.service.AlterationDriverAnnotationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class AlterationDriverAnnotationServiceImpl implements AlterationDriverAnnotationService {
+    
+    @Autowired
+    private AlterationDriverAnnotationRepository alterationDriverAnnotationRepository;
+    private final Set<String> NO_DATA_TIERS = new HashSet<>(Arrays.asList(null, "", "NA"));
+    
+    public CustomDriverAnnotationReport getCustomDriverAnnotationProps(List<String> molecularProfileIds) {
+        
+        List<AlterationDriverAnnotation> rows = alterationDriverAnnotationRepository
+            .getAlterationDriverAnnotations(molecularProfileIds);
+        
+        Set<String> tiers = rows.stream()
+            .filter(d -> !NO_DATA_TIERS.contains(d.getDriverTiersFilter()))
+            .map(AlterationDriverAnnotation::getDriverTiersFilter)
```

Comment:
Put the map first so you only have to call the getter once
```suggestion
            .map(AlterationDriverAnnotation::getDriverTiersFilter)
            .filter(driverTiersFilter -> !NO_DATA_TIERS.contains(driverTiersFilter))
```

Reply:
Ah yes, thanks for that suggestion. I never look at it that way. I'll keep that in mind in the future. So, updated!